Makefile invokes non-existent git command and fails

Trying to build the starter-kit on Debian, I hit errors trying to to build the code with `make`. Am I missing some prerequisite or using an insufficiently high version of some tool? If so, could this be added to the README?

My environment:

<details>
  <summary>environment details</summary>

* distro: Debian 9 (stretch)
* node version: v14.5.0
* npm version: 6.14.5
* git version: 2.11.0
* starter-kit commit: 080b20af94b10887f946d9bc98cd64f890bd02b1

</details>

I followed the steps in the README and this [blog post](https://cockpit-project.org/blog/cockpit-starter-kit.html), and got `make` errors.

### Steps to reproduce:

* `git clone https://github.com/cockpit-project/starter-kit.git`
* `cd starter-kit`
* `make`

### Expected results

Project should build without error.

### Actual results

`make` halts with the following error output:

```text
git restore --staged pkg/lib/patternfly
git: 'restore' is not a git command. See 'git --help'.

Did you mean this?
	remote
Makefile:161: recipe for target 'src/lib/patternfly/_fonts.scss' failed
```

<details>
  <summary>full make output</summary>

```text
$ make
/bin/sh: 1: rpmspec: not found
/bin/sh: 1: rpmspec: not found
/bin/sh: 1: rpmspec: not found
# if it exists already, npm install won't update it; force that so that we always get up-to-date packages
rm -f package-lock.json
# unset NODE_ENV, skips devDependencies otherwise
env -u NODE_ENV npm install
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated popper.js@1.16.1: You can find the new Popper v2 at @popperjs/core, this package is dedicated to the legacy v1
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated core-js@2.6.11: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.

> node-sass@4.14.1 install /opt/work/starter-kit/node_modules/node-sass
> node scripts/install.js

Downloading binary from https://github.com/sass/node-sass/releases/download/v4.14.1/linux-x64-83_binding.node
Download complete  ] - :
Binary saved to /opt/work/starter-kit/node_modules/node-sass/vendor/linux-x64-83/binding.node
Caching binary to /home/user/.npm/node-sass/4.14.1/linux-x64-83_binding.node

> core-js@2.6.11 postinstall /opt/work/starter-kit/node_modules/attr-accept/node_modules/core-js
> node -e "try{require('./postinstall')}catch(e){}"

Thank you for using core-js ( https://github.com/zloirock/core-js ) for polyfilling JavaScript standard library!

The project needs your help! Please consider supporting of core-js on Open Collective or Patreon: 
> https://opencollective.com/core-js 
> https://www.patreon.com/zloirock 

Also, the author of core-js ( https://github.com/zloirock ) is looking for a good job -)


> core-js@3.6.5 postinstall /opt/work/starter-kit/node_modules/core-js
> node -e "try{require('./postinstall')}catch(e){}"


> node-sass@4.14.1 postinstall /opt/work/starter-kit/node_modules/node-sass
> node scripts/build.js

Binary found at /opt/work/starter-kit/node_modules/node-sass/vendor/linux-x64-83/binding.node
Testing binary
Binary is fine
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN notsup Unsupported engine for watchpack-chokidar2@2.0.0: wanted: {"node":"<8.10.0"} (current: {"node":"14.5.0","npm":"6.14.5"})
npm WARN notsup Not compatible with your version of node/npm: watchpack-chokidar2@2.0.0
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@~2.1.2 (node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.2.7 (node_modules/watchpack-chokidar2/node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN po2json@1.0.0-beta-3 requires a peer of commander@^6.0.0 but none is installed. You must install peer dependencies yourself.

added 954 packages from 443 contributors and audited 956 packages in 187.395s

52 packages are looking for funding
  run `npm fund` for details

found 1 high severity vulnerability
  run `npm audit fix` to fix them, or `npm audit` for details
env -u NODE_ENV npm prune
npm WARN po2json@1.0.0-beta-3 requires a peer of commander@^6.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.1.3 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules/watchpack-chokidar2/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

audited 956 packages in 8.195s

52 packages are looking for funding
  run `npm fund` for details

found 1 high severity vulnerability
  run `npm audit fix` to fix them, or `npm audit` for details
git fetch --depth=1 https://github.com/cockpit-project/cockpit.git 227
remote: Enumerating objects: 1514, done.
remote: Counting objects: 100% (1514/1514), done.
remote: Compressing objects: 100% (1389/1389), done.
remote: Total 1500 (delta 253), reused 435 (delta 18), pack-reused 0
Receiving objects: 100% (1500/1500), 3.96 MiB | 1.19 MiB/s, done.
Resolving deltas: 100% (253/253), completed with 4 local objects.
From https://github.com/cockpit-project/cockpit
 * tag               227        -> FETCH_HEAD
mkdir -p pkg/lib/patternfly && git add pkg/lib/patternfly
git checkout --force FETCH_HEAD -- pkg/lib/patternfly
git restore --staged pkg/lib/patternfly
git: 'restore' is not a git command. See 'git --help'.

Did you mean this?
	remote
Makefile:161: recipe for target 'src/lib/patternfly/_fonts.scss' failed
make: *** [src/lib/patternfly/_fonts.scss] Error 1
```

</details>

### other comments

I assume the `rpmspec` command is some Red Hat–specific command I don't have on Debian, but lack of it doesn't seem to cause any fatal errors.

I was able to successfully build by patching the Makefile so that the `git restore` command is allowed to fail, but I don't yet know what downstream effects this might have.

My patch:

```patch
diff --git a/Makefile b/Makefile
index e28e42e..7836cc6 100644
--- a/Makefile
+++ b/Makefile
@@ -161,7 +161,7 @@ src/lib/patternfly/_fonts.scss:
        git fetch --depth=1 https://github.com/cockpit-project/cockpit.git 227
        mkdir -p pkg/lib/patternfly && git add pkg/lib/patternfly
        git checkout --force FETCH_HEAD -- pkg/lib/patternfly
-       git restore --staged pkg/lib/patternfly
+       -git restore --staged pkg/lib/patternfly
        mkdir -p src/lib && mv pkg/lib/patternfly src/lib/patternfly && rmdir -p pkg/lib
 
 # force serialization of the targets that call git, as they compete for the git lock

```

--- a/Makefile
+++ b/Makefile
@@ -161,7 +161,7 @@ src/lib/patternfly/_fonts.scss:
 	git fetch --depth=1 https://github.com/cockpit-project/cockpit.git 227
 	mkdir -p pkg/lib/patternfly && git add pkg/lib/patternfly
 	git checkout --force FETCH_HEAD -- pkg/lib/patternfly
-	git restore --staged pkg/lib/patternfly
+	git reset -- pkg/lib/patternfly
 	mkdir -p src/lib && mv pkg/lib/patternfly src/lib/patternfly && rmdir -p pkg/lib
 
 # force serialization of the targets that call git, as they compete for the git lock

